docs/data/data-grid/column-menu/ColumnMenuGridReferences.js

```diff
@@ -0,0 +1,250 @@
+  {
+    field: 'defaultComponent',
+    headerName: 'Default Components',
+    width: 280,
+    renderCell: (params) => <ComponentTag value={params.value} />,
+  },
+  { field: 'displayOrder', headerName: 'Display Order', width: 140, type: 'number' },
+];
+
+const rows = [
+  {
+    id: 1,
+    slot: 'ColumnMenuSortItem',
+    defaultComponent: 'GridColumnMenuSortItem',
+    design: 'Default',
+    displayOrder: 0,
+    plan: 'Community',
+  },
+  {
+    id: 2,
+    slot: 'ColumnMenuSortItem',
+    defaultComponent: 'GridColumnMenuSortItemSimple',
+    design: 'Simple',
+    displayOrder: 0,
+    plan: 'Community',
+  },
+  {
+    id: 3,
+    slot: 'ColumnMenuFilterItem',
+    defaultComponent: 'GridColumnMenuFilterItem',
+    design: 'Default',
+    displayOrder: 10,
+    plan: 'Community',
+  },
+  {
+    id: 4,
+    slot: 'ColumnMenuFilterItem',
+    defaultComponent: 'GridColumnMenuFilterItemSimple',
+    design: 'Simple',
+    displayOrder: 10,
+    plan: 'Community',
+  },
+  {
+    id: 5,
+    slot: 'ColumnMenuHideItem',
+    defaultComponent: 'GridColumnMenuHideItem',
+    design: 'Default',
+    displayOrder: 20,
+    plan: 'Community',
+  },
+  {
+    id: 6,
+    slot: 'ColumnMenuHideItem',
+    defaultComponent: 'GridColumnMenuHideItemSimple',
+    design: 'Simple',
+    displayOrder: 20,
+    plan: 'Community',
+  },
+  {
+    id: 7,
+    slot: 'ColumnMenuColumnsItem',
+    defaultComponent: 'GridColumnMenuColumnsItem',
+    design: 'Default',
+    displayOrder: 30,
+    plan: 'Community',
+  },
+  {
+    id: 8,
+    slot: 'ColumnMenuColumnsItem',
+    defaultComponent: 'GridColumnMenuColumnsItemSimple',
+    design: 'Simple',
+    displayOrder: 30,
+    plan: 'Community',
+  },
+  {
+    id: 9,
+    slot: 'ColumnMenuPinningItem',
+    defaultComponent: 'GridColumnMenuPinningItem',
+    design: 'Default',
+    displayOrder: 5,
+    plan: 'Pro',
+  },
+  {
+    id: 10,
+    slot: 'ColumnMenuPinningItem',
+    defaultComponent: 'GridColumnMenuPinningItemSimple',
+    design: 'Simple',
+    displayOrder: 35,
+    plan: 'Pro',
+  },
+  {
+    id: 11,
+    slot: 'ColumnMenuAggregationItem',
+    defaultComponent: 'GridColumnMenuAggregationItem',
+    design: 'Default',
+    displayOrder: 17,
+    plan: 'Premium',
+  },
+  {
+    id: 12,
+    slot: 'ColumnMenuAggregationItem',
+    defaultComponent: 'GridColumnMenuAggregationItemSimple',
+    design: 'Simple',
+    displayOrder: 37,
+    plan: 'Premium',
+  },
+  {
+    id: 13,
+    slot: 'ColumnMenuGroupingItem',
+    defaultComponent: 'GridColumnMenuRowGroupItem,GridColumnMenuRowUngroupItem',
+    design: 'Default',
+    displayOrder: 13,
+    plan: 'Premium',
+  },
+  {
+    id: 14,
+    slot: 'ColumnMenuGroupingItem',
+    defaultComponent:
+      'GridColumnMenuRowGroupItemSimple,GridColumnMenuRowUngroupItemSimple',
+    design: 'Simple',
+    displayOrder: 33,
+    plan: 'Premium',
+  },
+];
+
+export default function ColumnMenuGridReferences() {
```

Comment:
Maybe use plan icons with links instead of a column with chips?
<img width="829" alt="Screenshot 2022-12-01 at 12 22 35" src="https://user-images.githubusercontent.com/13808724/205040402-80708e1f-e578-4f16-80eb-caf10347f1cb.png">

I've also updated `ComponentTag` to match the style of code blocks in the markdown.
Here's a git diff if you find some changes in the screenshot above useful:

```diff
diff --git a/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx b/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx
index baae05d..68b13fe 100644
--- a/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx
+++ b/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx
@@ -5,42 +5,44 @@ import {
   useKeepGroupedColumnsHidden,
   GridRenderCellParams,
 } from '@mui/x-data-grid-premium';
-import Chip from '@mui/material/Chip';
-import Avatar from '@mui/material/Avatar';
 import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
 
-const getChipProperties = (plan: string) => {
+const getPlanProps = (plan: string) => {
   switch (plan.toLowerCase()) {
     case 'premium':
-      return { avatarLink: '/static/x/premium.svg', color: '#ffecc8' };
+      return {
+        href: '/x/introduction/licensing/#premium-plan',
+        className: 'plan-premium',
+        title: 'Premium plan',
+      };
     case 'pro':
-      return { avatarLink: '/static/x/pro.svg', color: '#c8e9ff' };
+      return {
+        href: '/x/introduction/licensing/#pro-plan',
+        className: 'plan-pro',
+        title: 'Pro plan',
+      };
     default:
-      return { avatarLink: undefined, color: '#c8ffdb' };
+      return null;
   }
 };
 
-function PlanTag(props: { plan?: string }) {
+function PlanIcon(props: { plan?: string }) {
   if (!props.plan) {
     return null;
   }
-  const chipPropperties = getChipProperties(props.plan);
-  const avatar = !chipPropperties.avatarLink ? undefined : (
-    <Avatar src={chipPropperties.avatarLink} />
-  );
+  const planProps = getPlanProps(props.plan);
+  if (!planProps) {
+    return null;
+  }
   return (
-    <Stack sx={{ width: '100%' }}>
-      <Chip
-        avatar={avatar}
-        sx={{ background: chipPropperties.color, color: 'rgba(0, 0, 0, 0.87)' }}
-        label={props.plan}
-      />
-    </Stack>
+    <a href={planProps.href} target="_blank" rel="noreferrer">
+      <span className={planProps.className} title={planProps.title} />
+    </a>
   );
 }
 
-function ComponentTag(props: { value?: string }) {
+function ComponentTag(props: { value?: string; plan?: string }) {
   if (!props.value) {
     return null;
   }
@@ -48,16 +50,22 @@ function ComponentTag(props: { value?: string }) {
   return (
     <Stack gap={0.5}>
       {components.map((c, key) => (
-        <Typography
-          key={key}
-          sx={{
-            borderRadius: '5px',
-            background: 'rgba(102, 178, 255, 0.15)',
-            fontSize: '0.8rem',
-          }}
-        >
-          {c}
-        </Typography>
+        <div>
+          <Typography
+            key={key}
+            sx={{
+              borderRadius: '5px',
+              background: 'rgba(102, 178, 255, 0.15)',
+              fontSize: '0.8rem',
+              fontFamily: 'Consolas, "Liberation Mono", Menlo, Courier, monospace',
+              padding: '0 5px',
+              display: 'inline-block',
+            }}
+          >
+            {c}
+          </Typography>
+          <PlanIcon plan={props.plan} />
+        </div>
       ))}
     </Stack>
   );
@@ -66,20 +74,10 @@ function ComponentTag(props: { value?: string }) {
 const columns = [
   {
     field: 'slot',
-    headerName: 'Slot',
-    width: 220,
+    headerName: 'Component',
+    width: 240,
     renderCell: (params: GridRenderCellParams<string>) => (
-      <ComponentTag value={params.value} />
-    ),
-  },
-  {
-    field: 'plan',
-    headerName: 'Plan',
-    width: 170,
-    type: 'singleSelect',
-    valueOptions: ['Premium', 'Pro', 'Community'],
-    renderCell: (params: GridRenderCellParams<string>) => (
-      <PlanTag plan={params.value} />
+      <ComponentTag value={params.value} plan={params.row.plan} />
     ),
   },
   { field: 'design', headerName: 'Menu Design' },
@@ -216,11 +214,6 @@ export default function ColumnMenuGridReferences() {
   const initialState = useKeepGroupedColumnsHidden({
     apiRef,
     initialState: {
-      columns: {
-        columnVisibilityModel: {
-          plan: false,
-        },
-      },
       rowGrouping: {
         model: ['design'],
       },
@@ -236,10 +229,7 @@ export default function ColumnMenuGridReferences() {
         hideFooter
         isGroupExpandedByDefault={(node) => node.groupingKey === 'Default'}
         groupingColDef={{
-          headerName: 'Design & Plan',
           hideDescendantCount: true,
-          leafField: 'plan',
-          width: 140,
         }}
       />
     </div>

```

Reply:
That's a really nice improvement, thank you 😍